staging/src/k8s.io/apimachinery/pkg/util/httpstream/wsstream/conn.go

```diff
@@ -272,6 +277,43 @@ func (conn *Conn) handle(ws *websocket.Conn) {
 	}
 }
 
+// handleV5 handles websocket data according to "V5" of the remotecommand subprotocol. The
+// V5 protocol defines a RESET signal as which is transmitted in the first byte. RESET will
+// close the stream, allowing other streams to finish transmitting.
+func (conn *Conn) handleV5(ws *websocket.Conn) {
+	defer conn.Close()
+	conn.initialize(ws)
+
+	for {
+		conn.resetTimeout()
+		var data []byte
+		if err := websocket.Message.Receive(ws, &data); err != nil {
```

Comment:
Does this mean in server `websocket.Message.Receive` handles ping frames?

Reply:
Antonio has pointed out that the `golang.org/x/net/websockets` library automatically sends the `PONG` reply to a `PING` here:

https://cs.opensource.google/go/x/net/+/refs/tags/v0.12.0:websocket/hybi.go;l=291-305

I have manually verified this :
```
I0713 14:37:50.169615 4081803 websocket.go:154] Starting Pings with 5000000000 period
I0713 14:37:50.169619 4081803 websocket.go:165] Websocket First Ping Sent...
I0713 14:37:50.169631 4081803 websocket.go:172] Websocket First Ping Sent...SUCCESSFUL
I0713 14:37:50.169733 4081803 websocket.go:205] Websocket Pong received...
```

So it appears we do not need to dig into the low-level `Codec` to implement the heartbeat.

Reply:
ping frames are control frames and not data frames, spdy implementation has a bug because considers them as data frames causing issues like https://github.com/moby/spdystream/pull/86